src/lib/FlightTasks/tasks/FlightTaskAuto.cpp

```diff
@@ -97,99 +103,110 @@ bool FlightTaskAuto::_evaluateTriplets()
-	// TODO This is a hack and it would be much better if the navigator only sends out a waypoints once tthey have changed.
+	if (!(fabsf(_triplet_target(0) - tmp_target(0)) > 0.001f || fabsf(_triplet_target(1) - tmp_target(1)) > 0.001f
+	      || fabsf(_triplet_target(2) - tmp_target(2)) > 0.001f)) {
+		// Nothing has changed: just keep old waypoints.
+		triplet_update = false;
 
-	// dont't do any updates if the current target has not changed
-	if (!(fabsf(target(0) - _target(0)) > 0.001f || fabsf(target(1) - _target(1)) > 0.001f
-	      || fabsf(target(2) - _target(2)) > 0.001f)) {
-		// nothing has changed: just keep old waypoints
-		return true;
-	}
+	} else {
+		_triplet_target = tmp_target;
+
+		if (!PX4_ISFINITE(_triplet_target(0)) || !PX4_ISFINITE(_triplet_target(1))) {
+			// Horizontal target is not finite.
+			_triplet_target(0) = _position(0);
+			_triplet_target(1) = _position(1);
+		}
 
-	// update all waypoints
-	_target = target;
+		if (!PX4_ISFINITE(_triplet_target(2))) {
+			_triplet_target(2) = _position(2);
+		}
 
-	if (!PX4_ISFINITE(_target(0)) || !PX4_ISFINITE(_target(1))) {
-		// Horizontal target is not finite. */
-		_target(0) = _position(0);
-		_target(1) = _position(1);
-	}
+		// If _triplet_target has updated, update also _triplet_prev_wp and _triplet_next_wp.
+		_prev_prev_wp = _triplet_prev_wp;
 
-	if (!PX4_ISFINITE(_target(2))) {
-		_target(2) = _position(2);
-	}
+		if (_isFinite(_sub_triplet_setpoint->get().previous) && _sub_triplet_setpoint->get().previous.valid) {
+			map_projection_project(&_reference_position, _sub_triplet_setpoint->get().previous.lat,
+					       _sub_triplet_setpoint->get().previous.lon, &_triplet_prev_wp(0), &_triplet_prev_wp(1));
+			_triplet_prev_wp(2) = -(_sub_triplet_setpoint->get().previous.alt - _reference_altitude);
+
+		} else {
+			_triplet_prev_wp = _position;
+		}
 
-	_prev_prev_wp = _prev_wp;
+		if (_type == WaypointType::loiter) {
+			_triplet_next_wp = _triplet_target;
 
-	if (_isFinite(_sub_triplet_setpoint->get().previous) && _sub_triplet_setpoint->get().previous.valid) {
-		map_projection_project(&_reference_position, _sub_triplet_setpoint->get().previous.lat,
-				       _sub_triplet_setpoint->get().previous.lon, &_prev_wp(0), &_prev_wp(1));
-		_prev_wp(2) = -(_sub_triplet_setpoint->get().previous.alt - _reference_altitude);
+		} else if (_isFinite(_sub_triplet_setpoint->get().next) && _sub_triplet_setpoint->get().next.valid) {
+			map_projection_project(&_reference_position, _sub_triplet_setpoint->get().next.lat,
+					       _sub_triplet_setpoint->get().next.lon, &_triplet_next_wp(0), &_triplet_next_wp(1));
+			_triplet_next_wp(2) = -(_sub_triplet_setpoint->get().next.alt - _reference_altitude);
 
-	} else {
-		_prev_wp = _position;
+		} else {
+			_triplet_next_wp = _triplet_target;
+		}
 	}
 
-	if (_type == WaypointType::loiter) {
-		_next_wp = _target;
+	// Check if yaw target is valid.
+	_yaw_setpoint = _sub_triplet_setpoint->get().current.yaw;
```

Comment:
the way the navigator currently runs, the update of the yaw_setpoint needs to happen continuously because the navigator implements the yaw-slewrate. Similar as cruise-speed, the yaw-setpoint needs to be updated at the beginning. 
Note, this will no longer be necessary with this https://github.com/PX4/Firmware/pull/10003, but that does not happen after this PR gets merged